### PR TITLE
temp: Fix `getArtifactsForBranchAndWorkflow` due to GH problems

### PR DESCRIPTION
**Pull request by billyvg**
Filtering workflows is currently broken due to GH, in the meantime we need to filter them client-side.

### CODE REVIEW

File: __tests__/api/getArtifactsForBranchAndWorkflow.test.ts [hunk 1]
```diff
@@ -16,9 +16,9 @@ test('it gets workflow runs and a branch and workflow id and then gets artifacts
     owner: 'getsentry',
     repo: 'sentry',
     workflow_id: 'acceptance.yml',
-    branch: 'main',
-    per_page: 10,
-    status: 'completed',
+    // branch: 'main',
+    per_page: 30,
+    // status: 'status',
```
Comment (billyvg) [inline __tests__/api/getArtifactsForBranchAndWorkflow.test.ts:21]: We'll want to keep this status change when we revert this change. This is more correct. (`status` is overloaded in the GH API, where it can filter either a workflow run's `status` *or* the `conclusion`).

File: src/api/getArtifactsForBranchAndWorkflow.ts [hunk 2]
```diff
@@ -61,8 +61,9 @@ export async function getArtifactsForBranchAndWorkflow(
       repo,
       // Below is typed incorrectly, it needs to be a string but typed as number
       workflow_id: (workflow_id as unknown) as number,
-      branch,
-      status: 'completed',
+      // XXX: GH broke filtering so we have to do it client-side for now
+      // branch,
+      // status: 'success',
```
Comment (billyvg) [inline src/api/getArtifactsForBranchAndWorkflow.ts:66]: We'll want to keep this status change when we revert this change. This is more correct. (`status` is overloaded in the GH API, where it can filter either a workflow run's `status` *or* the `conclusion`).

**Review by armenzg — APPROVED** [review]
🎉

**Comment by billyvg** [comment]
Test run here: https://github.com/getsentry/sentry/runs/6026143163?check_suite_focus=true
